crates/biome_css_analyze/src/lint/nursery/no_invalid_direction_in_linear_gradient.rs

```diff
@@ -0,0 +1,148 @@
+use biome_analyze::{context::RuleContext, declare_rule, Ast, Rule, RuleDiagnostic, RuleSource};
+use biome_console::markup;
+use biome_css_syntax::{CssFunction, CssParameter};
+use biome_rowan::AstNode;
+use lazy_static::lazy_static;
+use regex::Regex;
+
+declare_rule! {
+    /// Disallow non-standard direction values for linear gradient functions.
+    ///
+    /// A valid and standard direction value is one of the following:
+    /// - an angle
+    /// - to plus a side-or-corner (to top, to bottom, to left, to right; to top right, to right top, to bottom left, etc.)
+    ///
+    /// A common mistake (matching outdated non-standard syntax) is to use just a side-or-corner without the preceding to.
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```css,expect_diagnostic
+    /// .foo { background: linear-gradient(top, #fff, #000); }
+    /// ```
+    ///
+    /// ```css,expect_diagnostic
+    /// .foo { background: linear-gradient(45, #fff, #000); }
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```css
+    /// .foo { background: linear-gradient(to top, #fff, #000); }
+    /// ```
+    ///
+    /// ```css
+    /// .foo { background: linear-gradient(45deg, #fff, #000); }
+    /// ```
+    ///
+    pub NoInvalidDirectionInLinearGradient {
+        version: "next",
+        name: "noInvalidDirectionInLinearGradient",
+        language: "css",
+        recommended: true,
+        sources: &[RuleSource::Stylelint("function-linear-gradient-no-nonstandard-direction")],
+    }
+}
+
+lazy_static! {
+    pub static ref GET_PREFIX_REGEX: Regex = Regex::new(r"^(-\w+-)").unwrap();
+    pub static ref LINEAR_GRADIENT_FUNCTION_NAME: Regex =
+        Regex::new(r"^(?i)(-webkit-|-moz-|-o-|-ms-)?linear-gradient").unwrap();
+    pub static ref IN_KEYWORD: Regex = Regex::new(r"(?i)\bin\b").unwrap();
+    pub static ref ANGLE: Regex = Regex::new(r"^[\d.]+(?:deg|grad|rad|turn)$").unwrap();
+    pub static ref DIRECTION: Regex = Regex::new(r"(?i)top|left|bottom|right").unwrap();
+    pub static ref DIRECTION_WITH_TO: Regex = Regex::new(&format!(
+        r"(?i)^to ({})(?: ({}))?$",
+        DIRECTION.as_str(),
+        DIRECTION.as_str()
+    ))
+    .unwrap();
+    pub static ref DIRECTION_WITHOUT_TO: Regex = Regex::new(&format!(
+        r"(?i)^({})(?: ({}))?$",
+        DIRECTION.as_str(),
+        DIRECTION.as_str()
+    ))
+    .unwrap();
+    pub static ref DIGIT: Regex = Regex::new(r"[\d.]").unwrap();
+}
+
+impl Rule for NoInvalidDirectionInLinearGradient {
+    type Query = Ast<CssFunction>;
+    type State = CssParameter;
+    type Signals = Option<Self::State>;
+    type Options = ();
+
+    fn run(ctx: &RuleContext<Self>) -> Option<Self::State> {
```

Comment:
```suggestion
    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
```

Did you change this part?